crates/subspace-farmer/src/farming/tests.rs

```diff
@@ -0,0 +1,118 @@
+use crate::commitments::Commitments;
+use crate::farming::Farming;
+use crate::identity::Identity;
+use crate::mock_rpc::MockRpc;
+use crate::plot::Plot;
+use std::sync::Arc;
+use subspace_core_primitives::{Piece, Salt, Tag, TAG_SIZE};
+use subspace_rpc_primitives::SlotInfo;
+use tempfile::TempDir;
+use tokio::sync::mpsc;
+use tokio::time::{sleep, Duration};
+
+fn init() {
+    let _ = env_logger::builder().is_test(true).try_init();
+}
+
+async fn farming_simulator(slots: Vec<SlotInfo>, tags: Vec<Tag>) {
+    init();
+
+    let base_directory = TempDir::new().unwrap();
+
+    let piece: Piece = [9u8; 4096].into();
+    let salt: Salt = slots[0].salt.clone(); // the first slots salt should be used for the initial commitments
+    let index = 0;
+
+    let plot = Plot::open_or_create(&base_directory).await.unwrap();
+    let commitments = Commitments::new(base_directory.path().join("commitments").into())
+        .await
+        .unwrap();
+    plot.write_many(Arc::new(vec![piece]), index).await.unwrap();
+    commitments.create(salt, plot.clone()).await.unwrap();
+
+    let identity;
+    let res = Identity::open_or_create(&base_directory);
+    match res {
+        Ok(result) => identity = result,
+        Err(_) => panic!("Identity fail!"),
+    }
+
+    let (_metadata_sender, metadata_recv) = mpsc::channel(10);
+    let (_block_sender, block_recv) = mpsc::channel(10);
+    let (_newhead_sender, newhead_recv) = mpsc::channel(10);
+    let (slot_sender, slot_recv) = mpsc::channel(10);
+    let (tag_sender, tag_recv) = mpsc::channel(10);
+
+    let _ = slot_sender.send(slots[0].clone()).await; // send only the first slot, so that farmer can start
```

Comment:
I'd use an `unwrap` simply if this should not fail, same to the other cases below.
```suggestion
    slot_sender.send(slots[0].clone()).await.unwrap(); // send only the first slot, so that farmer can start
```